pkg/controller/controller.go

```diff
@@ -742,6 +758,19 @@ func (p *csiProvisioner) prepareProvision(ctx context.Context, claim *v1.Persist
 		deletionAnnSecrets.namespace = provisionerSecretRef.Namespace
 	}
 
+	if utilfeature.DefaultFeatureGate.Enabled(features.VolumeAttributesClass) && vacName != nil {
+		vac, err := p.client.StorageV1alpha1().VolumeAttributesClasses().Get(ctx, *vacName, metav1.GetOptions{})
+		if err != nil {
+			return nil, controller.ProvisioningNoChange, err
+		}
+
+		if vac.DriverName != p.driverName {
```

Comment:
If an error occurs, the pvc will stuck in Pending state. users how to know the reason and how to fix it? It seems that the error message won't be found in the pvc's status field. Is it possible to omit an event to the pvc?

Reply:
+1 to omit an event. We need to retry on this as well.

Reply:
Newest version records an event here if the PVC and VAC's driver names are mismatched. Existing code should drive a retry as normal with provisioning failures.

Reply:
Resolving this, reverted the change because of https://github.com/kubernetes-csi/external-provisioner/pull/1068#discussion_r1392414749